src/WingetCreateCLI/TokenHelper.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI
+{
+    using System;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using Windows.Win32;
+    using Windows.Win32.Foundation;
+    using Windows.Win32.Security.Credentials;
+
+    /// <summary>
+    /// Provides functionality for caching and retrieving the GitHub OAuth token.
+    /// </summary>
+    public static class TokenHelper
+    {
+        // Windows credentials manager
+        private const string CredTargetName = "git:https://aka.ms/winget-create";
+        private const string CredUserName = "Personal Access Token";
+
+        // Environment variable
+        private const string TokenEnvironmentVariable = "WINGET_CREATE_GITHUB_TOKEN";
+
+        /// <summary>
+        /// Deletes the token.
+        /// </summary>
+        /// <returns>True if the token was deleted, false otherwise.</returns>
+        public static bool Delete()
+        {
+            return PInvoke.CredDelete(CredTargetName, CRED_TYPE.CRED_TYPE_GENERIC);
+        }
+
+        /// <summary>
+        /// Reads the token.
+        /// </summary>
+        /// <param name="token">Output token.</param>
+        /// <returns>True if the token was read, false otherwise.</returns>
+        public static unsafe bool TryRead(out string token) => TryReadFromEnvironmentVariable(out token) || TryReadFromCredentialManager(out token);

```

Comment:
@denelon should we include a _temporary_ code for moving the token from the filesystem to Windows Credentials Manager and cleanup the token file?